src/PolyhedralGeometry/PolyhedralComplex/properties.jl

```diff
@@ -0,0 +1,431 @@
+
+julia> dim(PC)
+2
+```
+"""
+dim(PC::PolyhedralComplex) = Polymake.fan.dim(pm_object(PC))::Int
+
+@doc Markdown.doc"""
+    polyhedra_of_dim(PC::PolyhedralComplex, polyhedron_dim::Int)
+
+Return the polyhedra of a given dimension in the polyhedral complex `PC`.
+
+# Examples
+```jldoctest
+julia> IM = IncidenceMatrix([[1,2,3],[1,3,4]]);
+
+julia> VR = [0 0; 1 0; 1 1; 0 1];
+
+julia> PC = PolyhedralComplex(IM, VR);
+
+julia> P1s = polyhedra_of_dim(PC,1)
+5-element SubObjectIterator{Polyhedron}:
+ A polyhedron in ambient dimension 2
+ A polyhedron in ambient dimension 2
+ A polyhedron in ambient dimension 2
+ A polyhedron in ambient dimension 2
+ A polyhedron in ambient dimension 2
+
+julia> for p in P1s
+       println(dim(p))
+       end
+1
+1
+1
+1
+1
+```
+"""
+function polyhedra_of_dim(PC::PolyhedralComplex, polyhedron_dim::Int)
+    n = polyhedron_dim - lineality_dim(PC) + 1
+    n < 0 && return nothing
+    pfaces = Polymake.fan.cones_of_dim(pm_object(PC), n)
+    nfaces = Polymake.nrows(pfaces)
+    rfaces = Vector{Int64}()
+    nfarf = 0
+    farf = Polymake.to_one_based_indexing(pm_object(PC).FAR_VERTICES)
+    for index in 1:nfaces
+        face = Polymake.row(pfaces, index)
+        if face <= farf
+            nfarf += 1
+        else
+            append!(rfaces, index)
+        end
+    end
+    return SubObjectIterator{Polyhedron}(pm_object(PC), _ith_polyhedron, length(rfaces), (f_dim = n, f_ind = rfaces))
+end
+
+function _ith_polyhedron(::Type{Polyhedron}, PC::Polymake.BigObject, i::Base.Integer; f_dim::Int = -1, f_ind::Vector{Int64} = Vector{Int64}())
+    pface = Polymake.row(Polymake.fan.cones_of_dim(PC, f_dim), f_ind[i])
+    return Polyhedron(Polymake.polytope.Polytope(VERTICES = PC.VERTICES[collect(pface),:], LINEALITY_SPACE = PC.LINEALITY_SPACE))
+end
+
+lineality_space(PC::PolyhedralComplex) = SubObjectIterator{RayVector{Polymake.Rational}}(pm_object(PC), _lineality_polyhedron, lineality_dim(PC))
+
+lineality_dim(PC::PolyhedralComplex) = pm_object(PC).LINEALITY_DIM
+
+
+@doc Markdown.doc"""
+    f_vector(PC::PolyhedralComplex)
+
+Compute the vector $(f₀,f₁,f₂,...,f_{dim(PC))$` where $f_i$ is the number of
+faces of $PC$ of dimension $i$.
+
+# Examples
+```jldoctest
+julia> VR = [0 0; 1 0; -1 0; 0 1];
+
+julia> IM = IncidenceMatrix([[1,2,4],[1,3,4]]);
+
+julia> far_vertices = [2,3,4];
+
+julia> PC = PolyhedralComplex(IM, VR, far_vertices);
+
+julia> f_vector(PC)
+3-element Vector{Int64}:
+ 1
+ 3
+ 2
+```
+"""
+function f_vector(PC::PolyhedralComplex)
+    ldim = lineality_dim(PC)
+    f_vec=vcat(zeros(Int64, ldim), [length(polyhedra_of_dim(PC,i)) for i in ldim:dim(PC)])
+    return f_vec
+end
+
+
+@doc Markdown.doc"""
+    nrays(PC::PolyhedralComplex)
+
+Return the number of rays of `PC`.
+
+# Examples
+```jldoctest
+julia> VR = [0 0; 1 0; -1 0; 0 1];
+
+julia> IM = IncidenceMatrix([[1,2,4],[1,3,4]]);
+
+julia> far_vertices = [2,3,4];
+
+julia> PC = PolyhedralComplex(IM, VR, far_vertices);
+
+julia> nvertices(PC)
+1
```

Comment:
```suggestion
julia> nrays(PC)
3
```
example swapped with `nvertices`